marimo/_plugins/ui/_impl/tables/utils.py

```diff
@@ -46,7 +48,7 @@ def get_table_manager_or_none(data: Any) -> TableManager[Any] | None:
                 return manager(data)
 
     # If we have a DataFrameLike object, use the DataFrameProtocolTableManager
-    if isinstance(data, DataFrameLike):
+    if is_dataframe_like(data):
         return DataFrameProtocolTableManager(data)
```

Comment:
Maybe we can add a fall back to normal rendering approach if `__dataframe__` method itself returns a non-dataframe object or raises an exception.

Reply:
done, thanks for the suggestion!